eng/pipelines/templates/jobs/sdk-job-matrix.yml

```
@@ -79,6 +79,9 @@ parameters:
     publishXunitResults: true
   - categoryName: AoT
     runAoTTests: true
+  - categoryName: TestBuild
+    buildArchitecture: arm64
+    runtimeIdentifier: osx-arm64
```

Yanni won't be back until next week. What I see is that the ci/pr ymls set the helix target queues but it's the job matrix that actually breaks this out. He might have a better way of solving this but for now, I think the below should let you remove the duplication in the sdk-build.yml as it should override the value...I think. 
```suggestion
    runtimeIdentifier: osx-arm64
    ${{ if eg(variables['System.TeamProject'], 'public') }}:
      helixTargetQueue: osx.13.arm64.open
    ${{ else }}:
      helixTargetQueue: osx.13.arm64
```

---

I've added the legs at the pipeline-level so each pipeline has the correct Helix queue (PR pipeline has open, Official has non-open).